README.md

```diff
@@ -166,7 +166,7 @@ metadata:
   name: my-instrumentation
 spec:
   exporter:
-    endpoint: http://otel-collector:4317
+    endpoint: http://otel-collector-headless:4317
```

Comment:
Did you verify that this is indeed load balancing? Is this the recommended pattern for users?

Reply:
I did test this with OTEL javaagent. The agent does not set round_robin but that is another story for another repository. 

That said, there were people asking about load balancing in clients for other languages.

Reply:
I'm asking because if you were to set this to `dns:///otel-collector-headless:4317`, the gRPC client will get all the IPs from the DNS entry `otel-collector-headless` (which is an A record) and use them all. The default strategy is pick_first, which will just get one of those IPs. Upon failure, the next IP is used. When using a regular service, only one IP is resolved, which is the service that is backed by a Kubernetes service.

My point with all of this is that I don't understand the benefits that the headless service is bringing when using HTTP as the protocol instead of `dns:///`. I would have guessed that a regular service would work better when not using the DNS resolution, as it would be backed by a Kubernetes service that has the chance of redirecting requests in case of failures.

Reply:
Did you try using `dns:///` with OTEL auto-instrumentations? It might not even work.

Reply:
Apparently not https://github.com/open-telemetry/opentelemetry-java/blob/65e7145e9c6151416042757941d45f8970819248/sdk-extensions/autoconfigure/src/main/java/io/opentelemetry/sdk/autoconfigure/OtlpConfigUtil.java#L154

Reply:
Please submit an issue to spec repo `OTEL_EXPORTER_OTLP_ENDPOINT` should support `dns:///` prefix.

Reply:
I still don't understand why headless is better for this case if the client cannot take advantage of obtaining the backing IP addresses the service provides.